src/coroutine/memoset/query.rs

```diff
@@ -87,8 +88,13 @@ where
         store: &Store<F>,
         result: AllocatedPtr<F>,
         dependency_provenances: Vec<AllocatedPtr<F>>,
+        allocated_key: Option<&AllocatedPtr<F>>,
```

Comment:
I don't understand why you're passing an option here, since you only call this function once on a `Some`

Reply:
Because I wasn't sure if we would need to do this the other way. It's just for generality. Once we get this completely nailed down (and optimized), we can remove any unneeded code paths.